components/fabric8-utils/src/main/java/io/fabric8/utils/Zips.java

```diff
@@ -114,6 +114,15 @@ public static void unzip(InputStream in, File toDir) throws IOException {
                 if (!entry.isDirectory()) {
                     String entryName = entry.getName();
                     File toFile = new File(toDir, entryName);
+                    String fileDestinationFullPath = toFile.getPath();
+                    try{
+                    if (!fileDestinationFullPath.startsWith(toDir.getPath())); throw new IOException("Extracting results to different directory");
```

Comment:
Does this really fix the issue? because `fileDestinationFullPath` is getting created from `toDir` path only. 
Suspecting how `toFile` path could differ?

Reply:
```java
if(!toFile.getCanonicalPath().startsWith(toDir.getPath())) {
    new IOException("Extracting results to different directory");
}
```
will work.